### ISSUE TITLE
Unable to create hydrostatic.dat and kh.dat files

### ISSUE THREAD

**Issue by abhisheks865** [LHEEA/Nemoh#6]
Hello all,
I would appreciate if someone can help me with the below issue. In order to run Bemio, It needs hydrostatic.dat and kh.dat file from NEMOH. 
I tried to run the WEC-Sim BEMIO code (bemio.m) for Nemoh. It has four files in mesh folder (WEC-Sim-examples-BEMIO-NEMOH-RM3-Mesh) called hydrostatic_0.dat, hydrostatic_1.dat, KH_0.dat and KH_1.dat so it is running fine. But when I am trying to run NEMOH for my object then it is not creating above four files and due to that I am unable to run the code on my geometry. Is there any change needed in Nemoh.cal or Mesh.cal file to generate these files or somewhere else. I am sorry but I am just a beginner and finding difficulty in running NEMOH due to less documentation available.

Below are the files NEMOH code is generating in Mesh and folder.

![1](https://user-images.githubusercontent.com/71249618/98741813-ee86fd00-2372-11eb-9ce5-a1d11b82f628.JPG)

Below are the files NEMOH code is generating in Result folder.

![2](https://user-images.githubusercontent.com/71249618/98741833-f6df3800-2372-11eb-9ae2-dc7ef92a34d1.JPG)

And bemio.m code is giving me below error:

![3 error](https://user-images.githubusercontent.com/71249618/98741852-ff377300-2372-11eb-834f-5e2f9a2ca1e3.JPG)

Can somebody guide me how can I create hydrostatic.dat and kh.dat file from NEMOH?
Thanks in advance

**Comment by ccz359** [comment]
I am having the same issue, were you ever able to resolve this @abhisheks865 ?

**Comment by abhisheks865** [comment]
I had a hard time using NEMOH so I ended up doing the work on Ansys Aqwa.

Thanks,
Abhishek Sharma
+1(832)964-9436

ᐧ

**Comment by ruddykurnia** [comment]
Hello,

In the current versions of NEMOH, the hydrostatics output is produced only after the build-in mesh generator (Mesh) in NEMOH is executed.
If you are using Linux, attached codes (a Fortran source code and a compiled file) can be run for calculating the hydrostatics of an user-specified mesh.
[hydroNEMOH_linux.zip](https://github.com/LHEEA/Nemoh/files/7852162/hydroNEMOH_linux.zip)


Regards,
Ruddy

**Comment by gducrozet** [comment]
Previous comment solve the issue then.

**Comment by hfpim** [comment]
Hello, I'm having the same issue and I'm using Windows 10. How can I proceed?

**Comment by ccz359** [comment]
I ended up using Capytaine instead, but had to be very careful about the 3D file creation, specifically where the waterline and center of gravity was located. If you end up using meshmagick to make your files, the viewer in meshmagick is very helpful to verify that your files are coming out correct. I ran Capytaine using python (Spyder), it did take 2 hours to run using fairly simple 2 body WEC. Best of luck!

**Comment by hfpim** [comment]
@ccz359 did you get the "hydrostatics.dat" and "kh.dat" files with Capytaine? I used Capytaine and got the files, but I kept getting an error message when using WEC-Sim, which I could not debug.

**Comment by ccz359** [comment]
@hfpim Quick disclaimer: I last worked on this a year ago, so my memories are not super fresh on this stuff. I did get the hydrostatics and kh files using Capytaine. What were your error messages in WEC-Sim? Also, how long did it take your Capytaine to run? Because sometimes (if I remember correctly) it would produce empty files in a short run-time, which obviously weren't very helpful and produced errors in WEC-Sim.

**Comment by hfpim** [comment]
@ccz359 I actually got an error on Capytaine itself. I'm running into issues with complex numbers. netcdf format doesn't handle complex numbers and even though there is some information in the Capytaine tutorials page, I still can't figure the problem out. How did you handle that?

**Comment by ccz359** [comment]
@hfpim I'm not sure I ever did encounter those errors in Capytaine. I wrote a quick summary of the process I used last year and documented it in this pdf. Hopefully you find it useful (especially steps 5 thru 10). Good luck! And let me know if you have any questions.
[WECSim_Process_Manual_UNOFFICIAL.pdf](https://github.com/LHEEA/Nemoh/files/10114418/WECSim_Process_Manual_UNOFFICIAL.pdf)

**Comment by gducrozet** [comment]
For your information, a new Release of NEMOH will be provided by the end of the week here: https://gitlab.com/lheea/Nemoh

Hopefully, it will solve your issue but if it is not, you could open a new issue in this new repository (GitHub will not be maintained anymore).

**Comment by hfpim** [comment]
@ccz359 Thank you very much! It seems like I was going through a different path using Capytaine (at least according to your summary). I'll go over the whole summary and hopefully it will work fine. Thankfully I'm working with a simple sphere, so no overcomplicated geometries to deal with
